addons/mail/controllers/thread.py

```diff
@@ -71,6 +71,20 @@ def mail_thread_messages(self, thread_model, thread_id, fetch_params=None):
             "messages": messages.ids,
         }
 
+    @http.route("/mail/thread/pinned_messages", methods=["POST"], type="jsonrpc", auth="public", readonly=True)
+    @add_guest_to_context
+    def mail_thread_pins(self, thread_model, thread_id):
+        thread = self._get_thread_with_access(thread_model, thread_id, mode="read")
+        if not thread:
+            raise NotFound()
+        domain = [
+            ("res_id", "=", int(thread_id)),
+            ("model", "=", thread_model),
+            ("pinned_at", "!=", False),
+        ]
+        messages = request.env["mail.message"].search(domain).sorted(key="pinned_at", reverse=True)
```

Comment:
We might want to add id fallback to the sort to add determinism in the off chance 2 messages have the same pinned_at value.